app.ts

```diff
@@ -0,0 +1,378 @@
+			$("#tableWrapper").hide();
+			$("#loader").show();
+			this.searchedIndex = null;
+			this.searchMethod(searchValue);
+			$("#loader").hide();
+			$("#tableWrapper").show();
+		});
+
+		/** Handle previous page button click */
+		$("#prevPageButton").on("click", () => {
+			if ($("#prevPageButton").hasClass("hidden")) {
+				return;
+			}
+			$("#prevPageButton").addClass("hidden");
+			this.searchedIndex = null;
+			const firstRecordOfCurrentPage = (this.currentPage - 1) * this.recordsPerPage;
+			let fromRecord = firstRecordOfCurrentPage;
+			$("#nextPageButton").hide();
+			$("#prevPageButton").hide();
+			$("#tableWrapper").hide();
+			$("#loader").show();
+			this.totalRecords()
+				.then(() => {
+					$("#prevPageButton").removeClass("hidden");
+					$("#loader").hide();
+					$("#tableWrapper").show();
+					$("#nextPageButton").show();
+					$("#prevPageButton").show();
+					if (this.currentFirstRecordIndex <= 0) {
+						$("#prevPageButton").hide();
+					} else if (this.currentPage < this.totalPages) {
+						$("#nextPageButton").show();
+					}
+				})
+				.then(() => {
+					if (firstRecordOfCurrentPage <= this.recordsPerPage) {
+						this.currentPage = 1;
+						this.currentFirstRecordIndex = 0;
+					} else {
+						this.currentPage--;
+						this.currentFirstRecordIndex -= this.recordsPerPage;
+					}
+					fromRecord = this.currentFirstRecordIndex;
+					return this.displayData(fromRecord, this.recordsPerPage);
+				})
+				.catch((error) => {
+					throw error;
+				});
+		});
+
+		/** Handle next page button click */
+		$("#nextPageButton").on("click", () => {
+			if ($("#nextPageButton").hasClass("hidden")) {
+				return;
+			}
+			if (this.currentPage >= this.totalPages) {
+				const errorMessage = "Already on the last page";
+				window.alert(errorMessage);
+				throw new Error(errorMessage);
+			}
+			$("#nextPageButton").addClass("hidden");
+			this.searchedIndex = null;
+			let fromRecord = this.currentFirstRecordIndex;
+			$("#nextPageButton").hide();
+			$("#prevPageButton").hide();
+			$("#tableWrapper").hide();
+			$("#loader").show();
+			let totalRecCount: number | void;
+			this.totalRecords()
+				.then(() => {
+					$("#nextPageButton").removeClass("hidden");
+					$("#loader").hide();
+					$("#tableWrapper").show();
+					$("#nextPageButton").show();
+					$("#prevPageButton").show();
+					if (typeof totalRecCount === 'number' && this.currentFirstRecordIndex >= totalRecCount) {
+						$("#nextPageButton").hide();
+					}
+				})
+				.then(() => {
+					if (this.currentPage < this.totalPages) {
+						this.currentPage++;
+						this.currentFirstRecordIndex += this.recordsPerPage;
+					} else {
+						this.currentPage = this.totalPages;
+					}
+					fromRecord = this.currentFirstRecordIndex;
+					return this.displayData(fromRecord, this.recordsPerPage);
+				})
+				.catch((error) => {
+					throw error;
+				});
+		});
+	}
+}
+
+/** Initialize the app when the page loads */
+new InitializeApp();
```

Comment:
This feels out of place without the `window.onload = () => { ... };` mentioned in the readme of this project. Not saying your code has to look like that, but I would struggle to have something to comment about if you still used the example that we provided in the readme.